Source/Aptabase/Private/Aptabase.cpp

```diff
@@ -0,0 +1,26 @@
+﻿#include "Aptabase.h"
+
+#include <Modules/ModuleManager.h>
+
+#include "AptabaseAnalyticsProvider.h"
+
+TSharedPtr<IAnalyticsProvider> FAptabaseModule::CreateAnalyticsProvider(const FAnalyticsProviderConfigurationDelegate& GetConfigValue) const
+{
+	return AnalyticsProvider;
+}
+
+void FAptabaseModule::StartupModule()
+{
+	AnalyticsProvider = MakeShared<FAptabaseAnalyticsProvider>();
+}
+
+void FAptabaseModule::ShutdownModule()
+{
+	if (AnalyticsProvider.IsValid())
+	{
+		// TODO: Adjust the end session calls to be more in line with what other modules do: 1) end session on destruction & 2) end session if a new one is started
```

Comment:
how important is this?

Reply:
To give you a little bit of context, this refers to automatically calling the `EndSession` endpoint during shutdown in case the developer forgot to call it themselves.

I personally think it's quite important to have this before we merge it in, but I didn't find an implementation I like yet. I am still playing around with a few ideas before touching on the final.